core/storage/question/gae_models_test.py

```diff
@@ -294,16 +294,18 @@ def test_get_question_skill_links_by_skill_ids(self):
             )
         )
         self.assertEqual(len(question_skill_link_models), 2)
-        self.assertEqual(question_skill_link_models[0].skill_id, skill_id_2)
-        self.assertEqual(question_skill_link_models[1].skill_id, skill_id_1)
+        self.assertIn(
+            question_skill_link_models[0].skill_id, [skill_id_1, skill_id_2])
+        self.assertIn(
+            question_skill_link_models[1].skill_id, [skill_id_1, skill_id_2])
```

Comment:
While all of these sorting and 'assertIn' functionality are making existing tests deterministic, we haven't actually tested that the shuffling is working as expected. Apart from these tests I believe we need another test where we fix a seed for the randomizer and then check the order, and then change the seed for the randomizer and ensure we have some other order and set of questions

Reply:
I have added additional tests. PTAL.